lib/qif/writer.rb

```diff
@@ -61,11 +62,17 @@ def initialize(io, type = 'Bank', format = 'dd/mm/yyyy')
     
     # Add a transaction for writing
     def <<(transaction)
-      @transactions << transaction
+      case transaction.class.to_s
```

Comment:
I have also run across .QIF files with multiple account blocks. For example, in an export from Moneydance, there are a number of !Account blocks at the beginning of the file, one after another, describing/declaring each account, and then the !Account blocks recur later, followed by a !Type statement and then a bunch of transactions for that account.